File: tunnel-server/src/test/java/com/tuuzed/tunnel/server/TunnelServerTest.java
package com.tuuzed.tunnel.server;

import com.tuuzed.tunnel.common.interceptor.SimpleRequestInterceptor;
import com.tuuzed.tunnel.common.log4j.Log4jInitializer;
import com.tuuzed.tunnel.common.util.SslContexts;
import io.netty.handler.ssl.SslContext;
import org.apache.log4j.Level;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class TunnelServerTest {

    private TunnelServer tunnelServer;

    @Test
    public void start() throws Exception {
        tunnelServer.start();
        Thread.currentThread().join();
    }

    @Before
    public void setUp() throws Exception {
        Log4jInitializer.initializeThirdLibrary(Level.WARN);
        Log4jInitializer.builder().initialize();

        SimpleRequestInterceptor simpleRequestInterceptor = new SimpleRequestInterceptor(
            "tk123456", "1024-60000"
        );
        SslContext sslContext = SslContexts.forServer(
            "../resources/jks/server.jks",
            "stunnelpass",
            "stunnelpass"
        );
        this.tunnelServer = TunnelServer.builder()
            .setBossThreads(1)
            .setWorkerThreads(-1)
            .setProtoRequestInterceptor(simpleRequestInterceptor)

            .setBindAddr(null)
            .setBindPort(5000)

            .setSslEnable(true)
            .setSslContext(sslContext)
            .setSslBindAddr(null)
            .setSslBindPort(5001)

            .setHttpEnable(true)
            .setHttpBindAddr(null)
            .setHttpBindPort(80)
            .setHttpRequestInterceptor(simpleRequestInterceptor)

            .setHttpsEnable(true)
            .setHttpsContext(sslContext)
            .setHttpsBindAddr(null)
            .setHttpsBindPort(443)
            .setHttpsRequestInterceptor(simpleRequestInterceptor)

            .build();
    }

    @After
    public void shutDown() {
        this.tunnelServer.destroy();
    }


}

File: tunnel2-server/src/test/kotlin/tunnel2/server/TunnelServerTest.kt
package tunnel2.server

import org.apache.log4j.Level
import org.junit.After
import org.junit.Before
import org.junit.Test
import tunnel2.common.logging.LoggerFactory
import tunnel2.common.ssl.SslContexts
import tunnel2.server.interceptor.SimpleRequestInterceptor

class TunnelServerTest {
    private lateinit var tunnelServer: TunnelServer

    @Test
    @Throws(Exception::class)
    fun start() {
        tunnelServer.start()
        Thread.currentThread().join()
    }

    @Before
    @Throws(Exception::class)
    fun setUp() {
        LoggerFactory.configConsole(level = Level.WARN, names = *LoggerFactory.thirdLibs).apply()
        LoggerFactory.configConsole(level = Level.ALL).apply()
        LoggerFactory.configFile(level = Level.ALL, file = "../logs/t2s.log").apply()

        val simpleRequestInterceptor = SimpleRequestInterceptor(
            "tk123456", "1024-60000"
        )
        val sslContext = SslContexts.forServer(
            "../resources/jks/tunnels.jks",
            "stunnelpass",
            "stunnelpass"
        )
        this.tunnelServer = TunnelServer(
            tunnelRequestInterceptor = simpleRequestInterceptor,
            httpRequestInterceptor = simpleRequestInterceptor,
            httpsRequestInterceptor = simpleRequestInterceptor,

            sslEnable = true,
            sslContext = sslContext,

            httpEnable = true,
            httpsEnable = true,
            httpsContext = sslContext
        )
    }

    @After
    fun shutDown() {
        this.tunnelServer.destroy()
    }
}